CAIPs/caip-x.md

```diff
@@ -0,0 +1,99 @@
+    
+ **Optimized Resource Usage**: The `externally_connectable` method focuses on direct communication through message passing, which is generally more resource-efficient than script injection. This means less CPU and memory usage as the communication is handled by the browser’s built-in mechanisms rather than through additional JavaScript execution on the page.
+
+**Faster Page Load Times**: Without the overhead of injecting and running additional scripts on every page, pages load faster, providing a smoother and more responsive user experience. This performance improvement is critical for user retention and satisfaction, especially on resource-intensive applications or slower devices.
+
+#### So Why Wallet Discovery?
+
+Given the benefits of `externally_connectable` we need to solve for a key area where it fails to match the injected provider pattern.  
+`externally_connectable` requires that the webpage attempting to connect with the wallet know the extension's canonical extension id. Currently, there is no standard method for these wallets to provide their extension IDs to dApps. A standardized approach will improve interoperability, security, and user experience by ensuring that dApps can reliably identify and communicate with the intended wallet.
+
+Ideally browsers would eventually provide an API by which wallets can register the capabilities that they offer which webpages could access and choose to communicate with in a way that avoids the need for this additional discovery step.
+  
+### Specification
+
+**Extension ID Announcement Protocol:**
+
+Each wallet extension must implement a mechanism to announce its extension ID to a dApp. This should be done using a JavaScript event emitted by the wallet and listened for by the dApp. The event should contain the canonical extension ID as its payload.
+
+**Modification to dApp Listening Mechanisms:**
+
+dApps must modify their initialization routines to listen for the wallet's extension ID announcement event. Upon capturing the event, the dApp can store the extension ID and use it to establish a connection using the `externally_connectable` API.
+
+**Content Script Utilization**:
+
+While this CAIP significantly advances the way wallets interact with dApps through the `externally_connectable` API, it still necessitates the use of a content script. This script, however, requires fewer permissions compared to traditional methods and can be injected synchronously to facilitate the initial communication process. This adjustment is necessary to bridge the current browser limitations and set the stage for future enhancements.
```

Comment:
i think it's the same amount of permissions? just more lightweight code payload

Reply:
Worth mentioning somewhere that this is only meant as a temporary solution and should be superseded by any browser native discovery mechanism that is developed in the future?